### PR TITLE
Switch the FFI bindings to use the SQLite cryptostore

### PR DESCRIPTION
**Pull request by poljar**
Since we have no migration from Sled to SQLite we'll have to log our EX users out. cc @stefanceriu.

### CODE REVIEW

**Review by Hywan — APPROVED** [review]

File: bindings/matrix-sdk-ffi/src/client_builder.rs [hunk 1]
```diff
@@ -102,7 +103,24 @@ impl ClientBuilder {
...
+            let mut state_store =
+                matrix_sdk_sled::SledStateStore::builder().path(data_path.to_owned());
+
+            if let Some(passphrase) = builder.passphrase.as_deref() {
+                state_store = state_store.passphrase(passphrase.to_owned());
+            }
+
+            let state_store = state_store.build()?;
+
+            let crypto_store = RUNTIME.block_on(matrix_sdk_sqlite::SqliteCryptoStore::open(
```
Comment (stefanceriu) [inline bindings/matrix-sdk-ffi/src/client_builder.rs:115]: Maybe this method should be called something along the lines of `configureWithPath` if it does both creation and opening

Comment (stefanceriu) [inline bindings/matrix-sdk-ffi/src/client_builder.rs:115]: But also, why does the sled one have a builder and this one doesn't? 🤔

**Review by stefanceriu — APPROVED** [review]
🙌

File: bindings/matrix-sdk-ffi/src/client_builder.rs (hunk 1 above)
Comment (poljar) [reply bindings/matrix-sdk-ffi/src/client_builder.rs:115]: The state store is a bit more complex, i.e. you have the ability to expand the state store into a crypto store which uses the same underlying database.

In any case, once we have a SQLite state store as well, we'll add the ability to configure the SQLite store on the client builder and you won't have to access this low level interface.

Comment (poljar) [reply bindings/matrix-sdk-ffi/src/client_builder.rs:115]: When you say creation, you mean creation of the DB or the `SqliteCryptoStore` object? 

If the former, you can't create a SQLite DB without opening it. If the later what's the point of having the object if it's unable to do anything except call a `build()` or `open()` method.

Note that the open method is somewhat idiomatic in the SQLite context: https://docs.rs/rusqlite/latest/rusqlite/struct.Connection.html#method.open.

Comment (stefanceriu) [reply bindings/matrix-sdk-ffi/src/client_builder.rs:115]: > Note that the open method is somewhat idiomatic in the SQLite context

Right okay, that explains it

**Pull request merged**